common/views/components/SearchBar/SearchBar.tsx

```diff
@@ -29,19 +29,34 @@ const SearchInputWrapper = styled.div`
 const SearchButtonWrapper = styled.div`
   flex: 0 1 auto;
 `;
+type Props = {
+  placeholder: string;
+  id?: string;
+  isGlobalSearch?: boolean;
+};
 
-const SearchBar: FunctionComponent<{ placeholder: string }> = ({
+const SearchBar: FunctionComponent<Props> = ({
   placeholder,
+  id,
+  isGlobalSearch,
 }) => {
-  const { query } = useRouter();
-  const [inputQuery, setInputQuery] = useState((query.query as string) || '');
+  const router = useRouter();
+  const [inputQuery, setInputQuery] = useState(
+    (router?.query?.query as string) || ''
+  );
   const searchInput = useRef<HTMLInputElement>(null);
 
+  const handleKeyDown = (e: KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'Enter') {
+      document.getElementById('global-search-submit')?.click();
```

Comment:
This doesn't seem ideal from an accessibility perspective - is it possible to do the global search with a form (which should also help things work for no-JS users)?

Reply:
Also, if we stick with this approach, do you need to `document.getElementById`? Since we're in the same component can we either (a) just get a ref for the button or (b) probably the better option, use a [`router.push`](https://nextjs.org/docs/api-reference/next/router#usage) to achieve the same outcome as `click`ing the Link?

Reply:
Right, I've addressed this because it made a lot of sense. It did end up being a bit of a maze but I think it should all work nicely now.

Reply:
I tried to split the commits throughout the cleanup, these address it
https://github.com/wellcomecollection/wellcomecollection.org/pull/9234/commits/a00ef9d2a73643de90c473561e9dc9197311f475
https://github.com/wellcomecollection/wellcomecollection.org/pull/9234/commits/af927c4dc3fc1668983a9633e6809bb4efef6bbd